src/storage-types/src/sources.rs

```diff
@@ -2245,6 +2250,12 @@ impl RustType<ProtoPostgresSourcePublicationDetails> for PostgresSourcePublicati
                 .map(mz_postgres_util::desc::PostgresTableDesc::from_proto)
                 .collect::<Result<_, _>>()?,
             slot: proto.slot,
+            // The minimum value for timeline_id is 1, so we use 0 to indicate an unknown timeline
```

Comment:
I would just leave this in terms of concrete types and avoid dealing with magical sentinels.